src/App/Exercises/Js/HitTheMoleGame/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import './styles.css';
+import { SettingButton } from './components/SettingButton';
+
+export const HitTheMoleGame = () => {
+  const ChooseTime = [
+    { value: 1, label: '1 minuta', isActive: true },
+    { value: 2, label: '2 minuty', isActive: false },
+    { value: 3, label: '3 minuty', isActive: false },
+  ];
+  const ChooseMoles = [
+    { value: 1, label: '1 kret', isActive: true },
+    { value: 2, label: '2 krety', isActive: false },
+    { value: 3, label: '3 krety', isActive: false },
+  ];
+
+  const ActivSettings = (array) => {
+    const newArray = array.find((item) => item.isActive === true);
+    return newArray.value;
+  };
+  const [game, setGame] = useState(false);
+  const [gameSettings, setGameSettings] = useState({
```

Comment:
Kurcze, znów nie mam się do czego przyczepić 😄 oprócz..... 😃 tego miejsca ⬆️ 

czy będzie potrzeba kiedykolwiek użyć `setGameSettings`, czyli w domyśle w ten sposób zmienić `gameSettings` ? 🤔

Reply:
chciałam w ten sposób uaktualnić zmiane czasu i liczbę kretów po zmianie guzików i potem to przesłać do gry po kliknięciu start, tylko że gameSettings pokazuje mi undefinded i jak przesyłam w propsać setGameSettings i tam uaktulaniam stan to też mi wykrzuca błąd

Reply:
Widzę, że w dobrym kierunku idziesz więc nie podaję na tacy rozwiązania, bo wiem, że sama na to zaraz wpadniesz 😄

Reply:
O ile dobrze rozumiem, to jedno założenie jest nie do końca prawidłowe. Zadam pytanie pomocnicze: Czy potrzebujesz potem już w samej grze te array'ki? Czy może jednak tylko wartości, które zostały ustawione 🤔

Reply:
nie potrzebuje, potrzebuje tylko wydobyć aktywne value do gry, rozumiem że jest łatwiejsze rozwiąznie

Reply:
Ano właśnie. Ja bym zrobił stan (albo dwa) w głównym komponencie `HitTheMoleGame`, a ich settery przekazał do komponentu, gdzie manipulujesz aktywnością buttonów. Wtedy będziesz w stanie "ze środka" komponentu ustawić stan komponentu "rodzica". Tym sposobem będziesz miała ustawione settingsy w komponencie głównym, które będziesz mogła przekazać do jakiegoś przyszłego komponentu z grą 😃 Ja bym to tak zrobił 😄

Reply:
i dokładnie to próbuje zrobić i tym stanem miał być [gameSettings, setGameSettings] = useState i chce ten setter czyli setGameSettings przesłać do komponentu

Reply:
I nie działa to? Hmm zaraz pobiorę Twój branch i looknę na to. Chyba, że chcesz się zdzwonić i pogadamy o tym live?

Reply:
tak naprawde mi bez róznicy ale też mi się nie pali, spokojnie zasne i bez rozwiązania, więc bardziej jak ci wygodnie i jak masz czas

Reply:
Właściwie chyba będzie szybciej dzwoniąc. Ten link z zajęć na teamsach jest aktywny cały czas więc możesz wbijać 😃

Reply:
Prościej będzie po prostu zrobić dwa osobne state'y i przekazywać osobne settery do komponentów. Żeby to zrobić jak mówisz, to trzeba by jeszcze przekazać wartość tych settingsów i użyć spread operatora żeby sklejać ten obiekt 🤔 chyba niepotrzebne to 😃

Reply:
Mam pobrany Twój branch i popatrzę w kod, ale raczej przed zajęciami dzisiejszymi nic nie wymyślę. Po zajęciach spróbuję znaleźć przyczynę.

Reply:
timer poprawiony ale teraz coś kolory po kliknięciu na plansze źle działają na tą chwile

Reply:
Z tego co potestowałem to obstawiam, że te useState'y dla dobrego i złego kliknięcia źle działają. Może napiszę Ci założenia, jak gra ma działać. Wtedy będzie trzeba trochę logikę zmienić:

ETAP 1:
- na podstawie makiety: makieta_2 (z kretem) dodajemy planszę 5x2 (10 elementów w tablicy)
- dodajemy przyciski sterujące start oraz stop
- na kliknięcie w start plansza oraz przycisk stop się pojawiają zaś start znika
- na kliknięcie w stop przycisk start się pojawiają zaś start oraz plansza znika
- dodajemy kreta w losowym miejscu na planszy
- za każdym razem gdy ktoś kliknie start kret pojawia sie w innym miejscu
- kliknięcie w pole gdzie jest kret podświetla się na zielono na 300ms (makieta state_2)
- kliknięcie w pole gdzie gdzie kreta nie ma podświetla się na czerwono na 300ms (makieta state_3)
ETAP 2:
- dodajemy licznik punktów (jak na makiecie_2: wynik)
- kliknięcie w kreta powoduje dodanie 1 punktu
- nie trafienie w kreta powoduje odjęcie 1 punktu
- po kliknięciu w kreta kret pojawia sie w innym losowym miejscu
- po kliknięciu w stop ukazuje sie text z wynikiem podobny jak na makiecie_3, pole wynik się ukrywa (a pod spodem resetuje)
ETAP 3:
- tworzymy funkcjonalność, dzięki której kret powinien zmieniać miejsce swojego występowania na przyciskach w ustalonych odstępach czasowych (1000ms)
- po kliknięciu w kreta zmienia on swoje miejsce natychmiast oraz jego licznik czasu się resetuje
ETAP 4:
- Dodajemy sekcje przycisków czas-gry z makiety: makieta_1
- nie można wystartować gry bez wybranego czasu gry
- można wybrać tylko 1 czas gry
- wybrany czas gry się podświetla
- po wystartowaniu gry zaczyna się nam zliczać czas pole czas do końca z makieta_2
- jeśli czas upłynie pojawia się komunikat jak na makiecie 3 oraz stan gry się resetuje dokładnie tak samo jak w przypadku kliknięcia w stop
ETAP 5:
- Rozwijamy aplikację o możliwość wybrania ilości kretów na planszy. W momencie, gdy wybieramy:
 # 1 kreta - plansza ma rozmiar 10 kafelków (2 rzędy po 5) (1000ms kret zostaje na polu)
 # 2 krety - plansza ma rozmiar 15 kafelków (3 rzędy po 5) (500ms kret zostaje na polu)
 # 3 krety - plansza ma rozmiar 20 kafelków (4 rzędy po 5) (350ms kret zostaje na polu)
- Zasady gry pozostają niezmienne, jednak trzeba zwrócić uwagę na to, aby kret nie zajął miejsca, w którym znajduje się inny kret (zawsze musi być x kretów na planszy)
- Należy również pamiętać o tym, że każdy kret ma niezależny licznik.
- Gra nie może wystartować bez wybrania ilości kretów.

Reply:
Rozumiem, nie znałam zasad i pisałam bardziej jak mi się wydawało że powinno być, czyli kolory po kliknięciu niemają być w stacie tylko czasowo? a reszta kodu ma mniej więcej sens? Niby mi wszystko działa poza kolorami